server/src/feature_cache.rs

```diff
@@ -80,7 +80,7 @@ impl FeatureCache {
             .and_modify(|existing_features| {
                 existing_features.modify_in_place(delta);
             })
-            .or_insert(client_features);
+            .or_insert(client_features.modify_and_copy(delta));
```

Comment:
We can not put the values directly in, but need to run it through types method that parses events.

Reply:
Seems a bit unfortunate that we have to construct a new `ClientFeatures` just so modify_and_copy can clone it.

Running through the types code, it looks like `modify_in_place` and `modify_and_copy` are sharing the bulk of their code. Ideally the bulk of that method would be extracted to something that they could both share and a third method could be created that builds a `ClientFeatures` from a delta

Reply:
Good point, I thought about that also at one point, but did not want to make it complex. I will take a look now.

Reply:
Regarding the construction of new `ClientFeatures`, we need to construct it anyways at one point if the value is not in cache.